model/simulationtests/swimmingpool_indoor.rb

```diff
@@ -0,0 +1,138 @@
+poolMakeUpWaterSchedule.setName("MakeUpWaterSched")
+swimmingPoolIndoor.setMakeupWaterSupplySchedule(poolMakeUpWaterSchedule)
+
+# Cover Schedule Name
+poolCoverSchedule = OpenStudio::Model::ScheduleRuleset.new(model)
+poolCoverSchedule.setName("PoolCoverSched")
+poolCoverScheduleDay = poolCoverSchedule.defaultDaySchedule()
+poolCoverScheduleDay.addValue(OpenStudio::Time.new(0,6,0,0),0.5)
+poolCoverScheduleDay.addValue(OpenStudio::Time.new(0,20,0,0),0.0)
+poolCoverScheduleDay.addValue(OpenStudio::Time.new(0,24,0,0),0.5)
+swimmingPoolIndoor.setCoverSchedule(poolCoverSchedule)
+
+# Cover Evaporation Factor
+swimmingPoolIndoor.setCoverEvaporationFactor(0.8)
+
+# Cover Convection Factor
+swimmingPoolIndoor.setCoverConvectionFactor(0.2)
+
+# Cover Short-Wavelength Radiation Factor
+swimmingPoolIndoor.setCoverShortWavelengthRadiationFactor(0.9)
+
+# Cover Long-Wavelength Radiation Factor
+swimmingPoolIndoor.setCoverLongWavelengthRadiationFactor(0.5)
+
+# Pool Heating System Maximum Water Flow Rate {m3/s}
+swimmingPoolIndoor.setPoolHeatingSystemMaximumWaterFlowRate(0.1)
+
+# Pool Miscellaneous Equipment Power {W/(m3/s)}
+swimmingPoolIndoor.setPoolMiscellaneousEquipmentPower(0.6)
+
+# Setpoint Temperature Schedule
+poolSetpointTempSchedule = OpenStudio::Model::ScheduleRuleset.new(model, 27.0)
+poolSetpointTempSchedule.setName("PoolSetpointTempSched")
+swimmingPoolIndoor.setSetpointTemperatureSchedule(poolSetpointTempSchedule)
+
+# Maximum Number of People
+swimmingPoolIndoor.setMaximumNumberofPeople(15.0)
+
+# People Schedule
+poolOccupancySchedule = OpenStudio::Model::ScheduleRuleset.new(model)
+poolOccupancySchedule.setName("PoolOccupancySched")
+poolOccupancyScheduleDay = poolOccupancySchedule.defaultDaySchedule()
+poolOccupancyScheduleDay.addValue(OpenStudio::Time.new(0,6,0,0),0.0)
+poolOccupancyScheduleDay.addValue(OpenStudio::Time.new(0,9,0,0),1.0)
+poolOccupancyScheduleDay.addValue(OpenStudio::Time.new(0,11,0,0),0.5)
+poolOccupancyScheduleDay.addValue(OpenStudio::Time.new(0,13,0,0),1.0)
+poolOccupancyScheduleDay.addValue(OpenStudio::Time.new(0,16,0,0),0.5)
+poolOccupancyScheduleDay.addValue(OpenStudio::Time.new(0,20,0,0),1.0)
+poolOccupancyScheduleDay.addValue(OpenStudio::Time.new(0,24,0,0),0.0)
+swimmingPoolIndoor.setPeopleSchedule(poolOccupancySchedule)
+
+# People Heat Gain Schedule
+poolOccHeatGainSched = OpenStudio::Model::ScheduleRuleset.new(model, 300.0)
+poolOccHeatGainSched.setName("PoolOccHeatGainSched")
+swimmingPoolIndoor.setPeopleHeatGainSchedule(poolOccHeatGainSched)
+
+# Connect the pool to the heating loop
+heating_loop.addDemandBranchForComponent(swimmingPoolIndoor)
+
+
+###
+
+# add output reports
+add_out_vars = false
+if add_out_vars
+
+  freq = 'Detailed'
+
+  # Outputs implemented in the class
+  swimmingPoolIndoor.outputVariableNames.each do |varname|
+    outvar = OpenStudio::Model::OutputVariable.new(varname, model)
+    outvar.setReportingFrequency(freq)
+  end
```

Comment:
Tested resulting sql and I do have all 22 variables (and caught a mistake in the E+ I/O).

'Indoor Pool Water Heating Rate' for eg does have sensible values too.